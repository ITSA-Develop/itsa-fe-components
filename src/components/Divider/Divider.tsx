import { Divider as AntDivider, DividerProps } from 'antd';

export const Divider = ({ ...rest }: DividerProps) => {
	return <AntDivider {...rest} />;
};
