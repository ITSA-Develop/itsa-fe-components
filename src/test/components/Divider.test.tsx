import { render, screen } from '@testing-library/react';
import { Divider } from '../../components/Divider/Divider';

describe('Divider component', () => {
	it('renders without crashing', () => {
		const { container } = render(<Divider />);
		expect(document.querySelector('.ant-divider')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('renders vertical divider when type="vertical"', () => {
		const { container } = render(<Divider type="vertical" />);
		const divider = document.querySelector('.ant-divider');
		expect(divider).toHaveClass('ant-divider-vertical');
		expect(container).toMatchSnapshot();
	});

	it('renders dashed divider when dashed=true', () => {
		const { container } = render(<Divider dashed />);
		const divider = document.querySelector('.ant-divider');
		expect(divider).toHaveClass('ant-divider-dashed');
		expect(container).toMatchSnapshot();
	});

	it('renders with text when children provided', () => {
		const { container } = render(<Divider>Test Text</Divider>);
		expect(screen.getByText('Test Text')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
