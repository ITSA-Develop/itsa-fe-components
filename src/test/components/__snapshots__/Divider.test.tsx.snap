// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Divider component > renders dashed divider when dashed=true 1`] = `
<div>
  <div
    class="ant-divider css-dev-only-do-not-override-l9pxc0 ant-divider-horizontal ant-divider-dashed"
    role="separator"
  />
</div>
`;

exports[`Divider component > renders vertical divider when type="vertical" 1`] = `
<div>
  <div
    class="ant-divider css-dev-only-do-not-override-l9pxc0 ant-divider-vertical"
    role="separator"
  />
</div>
`;

exports[`Divider component > renders with text when children provided 1`] = `
<div>
  <div
    class="ant-divider css-dev-only-do-not-override-l9pxc0 ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center"
    role="separator"
  >
    <span
      class="ant-divider-inner-text"
    >
      Test Text
    </span>
  </div>
</div>
`;

exports[`Divider component > renders without crashing 1`] = `
<div>
  <div
    class="ant-divider css-dev-only-do-not-override-l9pxc0 ant-divider-horizontal"
    role="separator"
  />
</div>
`;
